### ISSUE TITLE
Load textures from a byte array

### ISSUE THREAD

**Issue by m0uka** [Facepunch/sbox-issues#462]
**What can't you do?**
Currently we can only load textures from the mounted (addon) read-only directory.

**How would you like it to work?**
It would be nice to have an option to load textures from a byte array or the data directory. I know we can download textures from the web, but if I want to stream textures via websockets, it's impossible

**Comment by Rohansi** [comment]
Is this to play videos? Because that'd be a horrible way to play videos

This could just be reading images from a byte array, which would solve your use case and allow for things like Rust-like paintable signs

**Comment by m0uka** [comment]
Yeah I know, your alternative is much better, didn't think of that, but video playback pretty much depends on #464, because we need a way how to decode images (could be also used for the signs).

**Comment by handsomematt** [comment]
You can already do this with Sandbox.TextureLoader.Image.Load which is wraps SKBitmap.Decode and returns a Sandbox.Texture.

```csharp
var stream = new MemoryStream( imgData.Data );
Texture tex = Sandbox.TextureLoader.Image.Load( stream );
```

**Comment by m0uka** [comment]
Didn't know of that, thanks! Closing.
